src/inline-svg.directive.ts

```diff
@@ -93,12 +94,23 @@ export class InlineSVGDirective implements OnInit, OnChanges, OnDestroy {
   private _insertSVG(): void {
     if (!isPlatformServer(this.platformId) && !this._supportsSVG) { return; }
 
-    // Check if a URL was actually passed into the directive
-    if (!this.inlineSVG) {
+    // Check if a URL/string was actually passed into the directive
+    if (!this.inlineSVG && !this.isSVGString) {
       this._fail('No URL passed to [inlineSVG]');
       return;
+    } else if (!this.inlineSVG && this.isSVGString) {
+      this._fail('No SVG string passed to [inlineSVG]');
+      return;
     }
 
+    if (this.isSVGString) {
+      this.insertSvgElement(this._svgElementFromString(this.inlineSVG));
```

Comment:
Can you refactor it so we get the SVG here, and then call `insertSvgElement` with it?